e2e/mock/server.go

```diff
@@ -3,39 +3,78 @@ package mock
 import (
 	"net"
 
-	"github.com/hyperledger/fabric-protos-go/orderer"
-	"github.com/hyperledger/fabric-protos-go/peer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
 type Server struct {
-	GrpcServer *grpc.Server
-	Listener   net.Listener
+	peers   []*Peer
+	orderer *Orderer
 }
 
-func (s *Server) Start() {
+func ConstructServer(peerN int, Credentials credentials.TransportCredentials) (server *Server, err error) {
 	blockC := make(chan struct{}, 1000)
-
-	p := &Peer{
-		BlkSize: 10,
-		TxC:     blockC,
+	server = &Server{}
```

Comment:
normally, we prepare all parts and assemble them altogether at very end. For example:
```go
var peers []*Peer
// construct peers here
...
// construct Orderer
...

return &Server{peers: peers, orderer: orderer}, nil
```